libr/core/cmd_debug.c

```diff
@@ -1651,6 +1651,20 @@ static int cmd_debug_map(RCore *core, const char *input) {
 					if (libname) {
 						char *cmd, *res;
 						const char *file = map->file? map->file: map->name;
+						char *newfile = NULL;
```

Comment:
the scope of newfile is in the block inside r_file_exists()

Reply:
there's a check for newfile down below when it's time to remove it.

btw, is it possible/wise to have 2 RCore instances side-by-side, one for the executable and one for a library?

Reply:
nope. dont do that, just use rbin and load that stuff inside the current rcore session.

Reply:
dmi just prints the lib symbols ... it shouldn't touch the current session in any way

Reply:
before this did try using a new RBinFile for `dmi <addr>` but then some code somewhere (probably the ELF plugin code) modified what appeared to be global state ... but will try again with perhaps a new RBin.

sorry not very familiar with the code